parquet-column/src/main/java/org/apache/parquet/io/api/RecordConsumer.java

```diff
@@ -126,10 +126,8 @@
   abstract public void addDouble(double value);
 
   /**
-   * NoOps by default
-   * Subclass class can implement its own flushing logic
+   * subclass class should implement its own flushing logic
    */
-  public void flush() {
-  }
+  abstract public void flush();
```

Comment:
@julienledem 
Sure. I've issued another commit reverting it back to be non-_abstract_. Is there any need we make it _abstract_ in a major release, like 2.0?